src/graphlayout.jl

```diff
@@ -36,6 +36,9 @@ function gradientalg(f, df, epochs, x0)
                 eta = 0.5 * eta
                 theta .+= dtheta
             end
+            if sum(abs.(dtheta)) < 1e-16
+                return theta
```

Comment:
Shouldn't 1e-16 be a symbolic constant? It would make it more clear what kind of threshold are we looking at.

Reply:
Good point. It's the minimum stape size.  I added a constant to the gradient algorithm code.

Reply:
Oops forgot that const doesn't work for local vars in Julia